src/docs/codelabs/extended-flex-widget.md

```diff
@@ -0,0 +1,597 @@
+
+## Image widget 
+
+The `Image` widget displays an image. You either can reference images using a URL,
+or you can include images inside your app package. Since DartPad can't package an image,
+the following example uses an image from the network.
+
+#### Example: Image widget
+{:.no_toc}
+{{site.alert.secondary}}
+  The following example displays an image using `Image`. 
+  The `Image.network` method takes a string parameter that contains the image's URL.
+
+  This example displays an image that's stored remotely on [GitHub](https://github.com/flutter/website/tree/master/examples/layout/sizing/images).
+
+  **1.** Click the **Run** button.  
+{{site.alert.end}}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=b6f3084800bd139cdb522b8858bb58b7&amp;fw=true&amp;split=60" width="100%" height="400px"></iframe>
+
+## Putting it all together 
+
+You've almost reached the end of this codelab.
+Why not test your knowledge by applying the techniques
+you've learned to build a UI that displays a business card!
+
+You'll break down the layout into parts, which is how you'd
+create a UI in the real world. This exercise can be broken
+into three parts. In part 1,
+you'll implement a `Column` containing a name, title, and icon. 
+In part 2, you'll wrap the `Column` in a `Row`, tweak the layout to look nice, 
+and add contact information. In part 3,
+you'll add another `Row` containing an address.  
+
+### Part 1
+{:.no_toc}
+
+This part contains two examples. 
+In the first example, you'll build a `Column`,
+which contains a name and title. 
+In the second example, you'll wrap your `Column` in a `Row`, 
+which contains an icon.
+By the end of Part 1, 
+your code displays a name, business title, and business icon. 
+
+#### Example: Create the name and title
+{:.no_toc}
+{{site.alert.secondary}}
+
+  Build a `Column` that contains the following widgets: 
+
+<ul markdown="1">
+  <li markdown="1">
+  One Text widget that contains the name "Flutter McFlutter" and
+  the `style` property set to `Theme.of(context).textTheme.headline`.  
+  </li>
+  <li markdown="1">
+  A second `Text` widget that contains the business title "Experienced Developer."
+
+  For your Column, set the:
+  </li>
+ 
+  <li markdown="1">`mainAxisSize` property to `MainAxisSize.min`
+  </li>
+  <li markdown="1">`crossAxisAlignment` property to `CrossAxisAlignment.start`.
+  </li>
+</ul>
+    
+{{site.alert.end}}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=30ccbe0fcf31cc10eafba3aea8ff0697&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+#### Exercise: Wrap the Column in a Row
+{:.no_toc}
+{{site.alert.secondary}}
+  
+
+  Wrap your `Column` in a `Row` that contains the following widgets:
+
+<ul markdown="1">
+  <li markdown="1">
+  An `Icon` widget set to `Icons.account_circle` with a size of 50 pixels.
```

Comment:
I'd just say something like this:

Wrap your Column in a Row, with an icon to the left of it.
The Row should look like this:

[img]

The Row should contain the following, not necessarily in this order:

* Your Column widget.
* The Icon wrapped in a Padding widget.
  Use the following code for the icon:
  ```
  Padding(
      padding: ...
  ) 
```

Reply:
Should we acknowledge that wrapping a widget in DartPad is hard? It's so much easier in an IDE...

Reply:
Is it because you use IDE shortcuts?

Reply:
Yes, iirc IDEs make it very easy to wrap an object in another object. It's beyond the scope of DartPad, but something IDEs can easily help with.

Reply:
Dartpad has this feature but it's hard to discover the shortcut now
https://github.com/dart-lang/dart-pad/issues/1094

Reply:
I got all excited, but it seems I already know about these DartPad shortcuts. Being able to indent is handy, but it doesn't do everything I was looking for.

Reply:
As to whether we usually introduce tips in codelabs... I would introduce any tips that are especially helpful for what they're trying to do. If I can't imagine doing it without the shortcut, then I'd definitely mention the shortcut.